src/scores/probability/crps_impl.py

```diff
@@ -817,6 +818,8 @@ def crps_for_ensemble(
         reduce_dims: Dimensions to reduce. Can be "all" to reduce all dimensions.
         preserve_dims: Dimensions to preserve. Can be "all" to preserve all dimensions.
         weights: Weights for calculating a weighted mean of individual scores.
+        decomposition: If True, returns the CRPS with underforecast and overforecast
```

Comment:
It is would probably be good to give an explicit formula in the docstring of this decomposition. The reason is that the `crps_cdf` implementation has a decomposition with with underforecast and overforecast penalties, but the underforecast penalty is `crps_cdf` is different from the proposed underforecast penalty in `crps_for_ensemble`. i.e., if you convert the ensemble to an ecdf and use `crps_cdf` to get the underforecast penalty, in most cases it will be different from the underforecast penalty given by `crps_for_ensemble`. I don't know whether the fact that the decompositions are different also needs to be pointed out. @tennlee may have a view on this.

Reply:
I do think a brief note would be useful in the docstring of both cdf and ensemble on the over/unders to the effect that the decomposition values are not intended to be identical between the two.